File: plot/src/main/java/net/fwitz/math/plot/binary/split/SplitComplexFunctionPanel.java
package net.fwitz.math.plot.binary.split;

import net.fwitz.math.binary.split.SplitComplex;
import net.fwitz.math.plot.binary.BinaryNumberFunctionPanel;

public class SplitComplexFunctionPanel extends BinaryNumberFunctionPanel<SplitComplex, SplitComplex> {
    SplitComplexFunctionPanel(SplitComplexFunctionPlot plot) {
        super(SplitComplex.class, SplitComplex.class, plot);
    }

    @Override
    protected SplitComplexFunctionRenderer<SplitComplex> createRenderer(int width, int height) {
        return new SplitComplexFunctionRenderer<>(
                this, valueType,
                width, height,
                SplitComplexFunctionValuesFilters.getInstance());
    }
}

File: plot/src/main/kotlin/net/fwitz/math/plot/binary/split/SplitComplexFunctionPanel.kt
package net.fwitz.math.plot.binary.split

import net.fwitz.math.plot.binary.BinaryNumberFunctionPanel
import net.fwitz.math.plot.binary.BinaryNumberFunctionRenderer
import net.fwitz.math.binary.split.SplitComplex

class SplitComplexFunctionPanel internal constructor(plot: SplitComplexFunctionPlot) :
    BinaryNumberFunctionPanel<SplitComplex, SplitComplex>(
        SplitComplex::class.java, SplitComplex::class.java, plot
    ) {
    override fun createRenderer(width: Int, height: Int): BinaryNumberFunctionRenderer<SplitComplex, SplitComplex> {
        return SplitComplexFunctionRenderer(
            this, valueType,
            width, height,
            SplitComplexFunctionValuesFilters
        )
    }
}